<!--
WARNING: Do _not_ update this file; it is automatically generated from software-respository.md.rsp.
-->



# Contributed Software Repositories

It is common that research groups maintain their own software stacks. Some of these groups make their stacks available to other cluster user community sparing others the effort having to install those software tools themselves.  This is often done via so called Linux module environments.  To get access to these repositories and their software, make sure to load them first, e.g.
```sh
module load CBI
```
Then use `module avail` to list what modules are availble (or see below).  Next, to actually get access to one or more of the shared software modules, use `module load <name>` or `module load <name>/<version>`.  Here are a few examples:
```sh
module load bowtie2
module load bowtie2/2.2.6
module load python r bwa
```

It is possible to enable a software repository and load specific software in one call, e.g.
```sh
module load CBI python r bwa
```

Below are 2 contributed software repositories, each providing a set of software tools.

<ul class="nav nav-pills">
  <li class="active"><a data-toggle="pill" href="#queues-CBI">CBI</a></li>
  <li><a data-toggle="pill" href="#queues-Sali">Sali</a></li>
</ul>
<div class="tab-content" style="margin-top: 1ex;">
<div id="queues-CBI" class="tab-pane fadein active">

<h2>Module Software Repository: CBI</h2>

Maintained by: Henrik Bengtsson, <a href="http://cbi.ucsf.edu">Computational Biology and Informatics</a><br>
Enable repository: <code>module load CBI</code><br>
<br>

<dl>
  <dt>bcl2fastq</dt>
  <dd>
    <strong>bcl2fastq: Illumina Conversion Software</strong><br>
bcl2fastq Conversion Software both demultiplexes data and converts BCL files generated by Illumina sequencing systems to standard FASTQ file formats for downstream analysis.  Example: <code>bcl2fastq --version</code><br>
    URL: <a href="https://support.illumina.com/sequencing/sequencing_software/bcl2fastq-conversion-software.html">https://support.illumina.com/sequencing/sequencing_software/bcl2fastq-conversion-software.html</a><br>
  Versions: <em>2.20.0</em>
  </dd>

  <dt>bowtie2</dt>
  <dd>
    <strong>Bowtie 2: A Fast and Sensitive Gapped Read Aligner</strong><br>
Bowtie 2 is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences.  Example: <code>bowtie2 --version</code><br>
    URL: <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">http://bowtie-bio.sourceforge.net/bowtie2/index.shtml</a><br>
  Versions: <em>2.3.5</em>
  </dd>

  <dt>bwa</dt>
  <dd>
    <strong>BWA: Burrows-Wheeler Aligner</strong><br>
Burrows-Wheeler Aligner (BWA) is a software package for mapping low-divergent sequences against a large reference genome, such as the human genome. Example: <code>bwa</code>.<br>
    URL: <a href="http://bio-bwa.sourceforge.net/">http://bio-bwa.sourceforge.net/</a><br>
  Versions: <em>0.7.17</em>
  </dd>

  <dt>byobu</dt>
  <dd>
    <strong>byobu: Elegant Enhancement of the Otherwise Functional, Plain, Practical GNU Screen</strong><br>
Byobu is an elegant enhancement of the otherwise functional, plain, practical GNU Screen. Byobu includes an enhanced profile, configuration utilities, and system status notifications for the GNU screen window manager as well as the Tmux terminal multiplexer. Example: <code>byobu --version</code><br>
    URL: <a href="http://byobu.org">http://byobu.org</a>, <a href="https://github.com/dustinkirkland/byobu">https://github.com/dustinkirkland/byobu</a><br>
  Versions: <em>5.127</em>
  </dd>

  <dt>CBI-testing</dt>
  <dd>
    <strong>The Computational Biology and Informatics (CBI) Software Repository - TESTING ONLY!</strong><br>
This repository provides prototypical environment modules that can change at any time. They will live in this CBI-testing repository until they have proven to work and be stable - only then they will be considered for the main CBI repository.  WARNING: Use at your own risk.<br>
    URL: <a href="http://cbi.ucsf.edu/">http://cbi.ucsf.edu/</a><br>
  Versions: 
  </dd>

  <dt>cellranger</dt>
  <dd>
    <strong>Cell Ranger: 10x Genomics Pipeline for Single-Cell Data Analysis</strong><br>
Cell Ranger is a set of analysis pipelines that process Chromium Single Cell 3' RNA-seq output to align reads, generate gene-cell matrices and perform clustering and gene expression analysis.  Example: <code>cellranger</code><br>
    URL: <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger">https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger</a><br>
  Versions: 2.1.0, <em>3.0.2</em>
  </dd>

  <dt>emacs</dt>
  <dd>
    <strong>GNU Emacs: An Extensible, Customizable, Free/Libre Text Editor</strong><br>
At its core is an interpreter for Emacs Lisp, a dialect of the Lisp programming language with extensions to support text editing. Example: <code>emacs --version</code> and <code>emacs -nw</code>.<br>
    URL: <a href="https://www.gnu.org/software/emacs/">https://www.gnu.org/software/emacs/</a><br>
  Versions: 25.3, <em>26.1</em>
  </dd>

  <dt>gatk</dt>
  <dd>
    <strong>Genome Analysis Toolkit (GATK): Variant Discovery in High-Throughput Sequencing Data</strong><br>
Developed in the Data Sciences Platform at the Broad Institute, the toolkit offers a wide variety of tools with a primary focus on variant discovery and genotyping. Its powerful processing engine and high-performance computing features make it capable of taking on projects of any size.  Example: <code>gatk --help</code> and <code>gatk --list</code>.<br>
    URL: <a href="https://software.broadinstitute.org/gatk/">https://software.broadinstitute.org/gatk/</a>, <a href="https://github.com/broadinstitute/gatk/">https://github.com/broadinstitute/gatk/</a><br>
  Versions: <em>4.1.0.0</em>
  </dd>

  <dt>git-flow</dt>
  <dd>
    <strong>git-flow: Git Extension Git Flow (AVH Edition)</strong><br>
A collection of Git extensions to provide high-level repository operations for Vincent Driessen's branching model. Example: <code>git flow</code>.<br>
    URL: <a href="https://github.com/petervanderdoes/gitflow-avh">https://github.com/petervanderdoes/gitflow-avh</a>, <a href="https://github.com/nvie/gitflow">https://github.com/nvie/gitflow</a><br>
  Versions: <em>1.12.1</em>
  </dd>

  <dt>htop</dt>
  <dd>
    <strong>htop - An Interactive Process Viewer for Unix</strong><br>
<code>htop</code> is an interactive process viewer for Unix systems. It is a text-mode application (for console or X terminals) and requires ncurses. Example: <code>htop</code><br>
    URL: <a href="http://hisham.hm/htop/">http://hisham.hm/htop/</a><br>
  Versions: <em>2.2.0</em>
  </dd>

  <dt>htslib</dt>
  <dd>
    <strong>HTSlib: C Library for High-Throughput Sequencing Data Formats</strong><br>
HTSlib is an implementation of a unified C library for accessing common file formats, such as SAM, CRAM and VCF, used for high-throughput sequencing data, and is the core library used by samtools and bcftools. HTSlib also provides the bgzip, htsfile, and tabix utilities.  Example: <code>bgzip --version</code>, <code>htsfile --version</code>, and <code>tabix --version</code>.<br>
    URL: <a href="http://www.htslib.org/">http://www.htslib.org/</a><br>
  Versions: <em>1.9</em>
  </dd>

  <dt>kallisto</dt>
  <dd>
    <strong>kallisto: Near-optimal RNA-Seq Quantification</strong><br>
kallisto is a program for quantifying abundances of transcripts from RNA-Seq data, or more generally of target sequences using high-throughput sequencing reads. It is based on the novel idea of pseudoalignment for rapidly determining the compatibility of reads with targets, without the need for alignment.  Example: <code>kallisto version</code>.<br>
    URL: <a href="https://pachterlab.github.io/kallisto/about.html">https://pachterlab.github.io/kallisto/about.html</a>, <a href="https://github.com/pachterlab/kallisto">https://github.com/pachterlab/kallisto</a><br>
  Versions: 0.45.0, <em>0.45.1</em>
  </dd>

  <dt>pindel</dt>
  <dd>
    <strong>pindel: Detection of Indels and Structural Variations</strong><br>
Pindel can detect breakpoints of large deletions, medium sized insertions, inversions, tandem duplications and other structural variants at single-based resolution from next-gen sequence data. It uses a pattern growth approach to identify the breakpoints of these variants from paired-end short reads.  Example: <code>pindel</code>.<br>
    URL: <a href="https://www.sanger.ac.uk/science/tools/pindel">https://www.sanger.ac.uk/science/tools/pindel</a>, <a href="https://github.com/genome/pindel">https://github.com/genome/pindel</a><br>
  Versions: <em>0.2.5b8</em>
  </dd>

  <dt>qtop</dt>
  <dd>
    <strong>qtop: Monitor the State of Queueing Systems, Along with Related Information Relevant on HPC &amp; Grid Clusters</strong><br>
<code>qtop</code> (pronounced queue-top) is a tool written in order to monitor the state of Queueing Systems, along with related information relevant on HPC &amp; grid clusters. At present it supports PBS, SGE &amp; OAR families. Please help to increase that list in the Python version of the tool, qtop.py! Examples: <code>qtop</code> and <code>qtop -FGW</code>.<br>
    URL: <a href="https://github.com/qtop/qtop">https://github.com/qtop/qtop</a><br>
  Versions: <em>0.9.20161222</em>
  </dd>

  <dt>r</dt>
  <dd>
    <strong>R: The R Programming Language</strong><br>
The R programming language. Examples: <code>R --version</code> and <code>Rscript --version</code>.<br>
    URL: <a href="https://www.r-project.org/">https://www.r-project.org/</a><br>
  Versions: 3.2.0, 3.3.0, 3.4.0, 3.5.0, 3.5.3, <em>3.6.0</em>
  </dd>

  <dt>samtools</dt>
  <dd>
    <strong>SAMtools: Tools (written in C using htslib) for Manipulating Next-Generation Sequencing Data</strong><br>
SAMtools is a suite of programs for interacting with high-throughput sequencing data.  Example: <code>samtools --version</code>.<br>
    URL: <a href="http://www.htslib.org/">http://www.htslib.org/</a><br>
  Versions: <em>1.9</em>
  </dd>

  <dt>scl-python</dt>
  <dd>
    <strong>NA</strong><br>
Enables the CentOS Software Collection (SCL) <code>rh-python36</code> in the current environment.  This is an alternative to calling <code>source scl_source enable rh-python36</code>, which is not officially supported by RedHat/CentOS.  Example: <code>python --version</code> and <code>pip --version</code>.<br>
    URL: <a href="https://www.softwarecollections.org/en/scls/rhscl/rh-python36/">https://www.softwarecollections.org/en/scls/rhscl/rh-python36/</a><br>
  Versions: .incl/33, .incl/34, .incl/36, 3.3, 3.4, <em>3.6</em>
  </dd>

  <dt>shellcheck</dt>
  <dd>
    <strong>ShellCheck: A Shell Script Static Analysis Tool</strong><br>
ShellCheck finds bugs in your shell scripts.  Example: <code>shellcheck --version</code> and <code>shellcheck -x ~/.bashrc</code>.<br>
    URL: <a href="https://www.shellcheck.net/">https://www.shellcheck.net/</a>, <a href="https://github.com/koalaman/shellcheck/">https://github.com/koalaman/shellcheck/</a><br>
  Versions: <em>0.6.0</em>
  </dd>

  <dt>star</dt>
  <dd>
    <strong>STAR: Spliced Transcripts Alignment to a Reference</strong><br>
STAR (Spliced Transcripts Alignment to a Reference) is a fast NGS read aligner for RNA-seq data. Example: <code>STAR --help</code>.<br>
    URL: <a href="https://github.com/alexdobin/STAR">https://github.com/alexdobin/STAR</a><br>
  Versions: <em>2.7.0e</em>
  </dd>

  <dt>tmux</dt>
  <dd>
    <strong>tmux: A Terminal Multiplexer</strong><br>
tmux is a terminal multiplexer. It lets you switch easily between several programs in one terminal, detach them (they keep running in the background) and reattach them to a different terminal. And do a lot more.  Example: <code>tmux</code> and <code>man tmux</code>.<br>
    URL: <a href="https://github.com/tmux/tmux/wiki">https://github.com/tmux/tmux/wiki</a><br>
  Versions: <em>2.8</em>
  </dd>

  <dt>tophat</dt>
  <dd>
    <strong>TopHat: A Spliced Read Mapper for RNA-Seq</strong><br>
TopHat is a fast splice junction mapper for RNA-Seq reads. It aligns RNA-Seq reads to mammalian-sized genomes using the ultra high-throughput short read aligner Bowtie, and then analyzes the mapping results to identify splice junctions between exons.  Example: <code>tophat --version</code>.<br>
    URL: <a href="https://ccb.jhu.edu/software/tophat/index.shtml">https://ccb.jhu.edu/software/tophat/index.shtml</a>, <a href="https://github.com/infphilo/tophat">https://github.com/infphilo/tophat</a><br>
  Versions: <em>2.1.1</em>
  </dd>

  <dt>tree</dt>
  <dd>
    <strong>tree: List Content of Directories in a Tree-like Format</strong><br>
Tree is a recursive directory listing command that produces a depth indented listing of files, which is colorized ala dircolors if the <code>LS_COLORS</code> environment variable is set and output is to tty.. Example: <code>tree --help</code>.<br>
    URL: <a href="http://mama.indstate.edu/users/ice/tree/">http://mama.indstate.edu/users/ice/tree/</a>,<br>
  Versions: <em>1.8.0</em>
  </dd>

</dl>
</div> 

<div id="queues-Sali" class="tab-pane fade">

<h2>Module Software Repository: Sali</h2>

Maintained by: Joshua Baker-LePain, <a href="https://salilab.org/">Sali Lab Software Repository</a><br>
Enable repository: <code>module load Sali</code><br>
<br>

<dl>
  <dt>allosmod</dt>
  <dd>
  Versions: 
  </dd>

  <dt>amber</dt>
  <dd>
  Versions: <em>11</em>
  </dd>

  <dt>anaconda</dt>
  <dd>
  Versions: 
  </dd>

  <dt>blast</dt>
  <dd>
  Versions: <em>2.2.26</em>
  </dd>

  <dt>blast+</dt>
  <dd>
  Versions: 2.2.25, <em>2.2.28</em>
  </dd>

  <dt>boost</dt>
  <dd>
  Versions: <em>1.68.0</em>
  </dd>

  <dt>cgal</dt>
  <dd>
  Versions: <em>4.12.1</em>
  </dd>

  <dt>cmake</dt>
  <dd>
  Versions: <em>3.12.2</em>
  </dd>

  <dt>concavity</dt>
  <dd>
  Versions: <em>0.1</em>
  </dd>

  <dt>cryptosite</dt>
  <dd>
  Versions: 
  </dd>

  <dt>cuda</dt>
  <dd>
  Versions: 6.0.37, 7.5.18, 8.0.61, 9.0.176, <em>10.0.130</em>
  </dd>

  <dt>Cython</dt>
  <dd>
  Versions: <em>0.25.2</em>
  </dd>

  <dt>doxygen</dt>
  <dd>
  Versions: 1.8.6, <em>1.8.15</em>
  </dd>

  <dt>dssp</dt>
  <dd>
  Versions: 2.0.4, <em>2.2.1</em>
  </dd>

  <dt>eigen</dt>
  <dd>
  Versions: <em>3.3.5</em>
  </dd>

  <dt>eman</dt>
  <dd>
  Versions: 2.12, <em>2.2</em>
  </dd>

  <dt>featuresketch</dt>
  <dd>
  Versions: <em>3.0</em>
  </dd>

  <dt>fpocket</dt>
  <dd>
  Versions: <em>2.0</em>
  </dd>

  <dt>gcc</dt>
  <dd>
  Versions: 5.1.1, 6.4.1, <em>7.3.1</em>
  </dd>

  <dt>ghostscript</dt>
  <dd>
  Versions: <em>8.70</em>
  </dd>

  <dt>gnuplot</dt>
  <dd>
  Versions: <em>5.0.5</em>
  </dd>

  <dt>hdf5</dt>
  <dd>
  Versions: 1.8.14, 1.8.17, <em>1.10.1</em>
  </dd>

  <dt>ImageMagick</dt>
  <dd>
  Versions: <em>6.8.8.10</em>
  </dd>

  <dt>imp</dt>
  <dd>
  Versions: last_ok_build-i386, last_ok_build-x86_64, last_ok_build, nightly-i386, nightly-x86_64, nightly, 2.0.0-i386, 2.0.0-x86_64, 2.0.0, 2.0.1-i386, 2.0.1-x86_64, 2.0.1, 2.1.0-i386, 2.1.0-x86_64, 2.1.0, 2.1.1-i386, 2.1.1-x86_64, 2.1.1, 2.2.0-i386, 2.2.0-x86_64, 2.2.0, 2.2.1-i386, 2.2.1-x86_64, 2.2.1, 2.3.0-i386, 2.3.0-x86_64, 2.3.0, 2.3.1-i386, 2.3.1-x86_64, 2.3.1, 2.4.0-i386, 2.4.0-x86_64, 2.4.0, 2.5.0-i386, 2.5.0-x86_64, 2.5.0, 2.6.0-i386, 2.6.0-x86_64, 2.6.0, 2.6.1-i386, 2.6.1-x86_64, 2.6.1, 2.6.2-i386, 2.6.2-x86_64, 2.6.2, 2.7.0-i386, 2.7.0-x86_64, 2.7.0, 2.8.0-i386, 2.8.0-x86_64, 2.8.0, 2.9.0-i386, 2.9.0-x86_64, 2.9.0, 2.10.0-i386, 2.10.0-x86_64, 2.10.0, 2.10.1-i386, 2.10.1-x86_64, <em>2.10.1</em>
  </dd>

  <dt>imp-fast</dt>
  <dd>
  Versions: last_ok_build-i386, last_ok_build-x86_64, last_ok_build, nightly-i386, nightly-x86_64, nightly, 2.0.0-i386, 2.0.0-x86_64, 2.0.0, 2.0.1-i386, 2.0.1-x86_64, 2.0.1, 2.1.0-i386, 2.1.0-x86_64, 2.1.0, 2.1.1-i386, 2.1.1-x86_64, 2.1.1, 2.2.0-i386, 2.2.0-x86_64, 2.2.0, 2.2.1-i386, 2.2.1-x86_64, 2.2.1, 2.3.0-i386, 2.3.0-x86_64, 2.3.0, 2.3.1-i386, 2.3.1-x86_64, 2.3.1, 2.4.0-i386, 2.4.0-x86_64, 2.4.0, 2.5.0-i386, 2.5.0-x86_64, 2.5.0, 2.6.0-i386, 2.6.0-x86_64, 2.6.0, 2.6.1-i386, 2.6.1-x86_64, 2.6.1, 2.6.2-i386, 2.6.2-x86_64, 2.6.2, 2.7.0-i386, 2.7.0-x86_64, 2.7.0, 2.8.0-i386, 2.8.0-x86_64, 2.8.0, 2.9.0-i386, 2.9.0-x86_64, 2.9.0, 2.10.0-i386, 2.10.0-x86_64, 2.10.0, 2.10.1-i386, 2.10.1-x86_64, <em>2.10.1</em>
  </dd>

  <dt>libtau</dt>
  <dd>
  Versions: <em>1.0.1</em>
  </dd>

  <dt>main</dt>
  <dd>
  Versions: <em>2013</em>
  </dd>

  <dt>matlab</dt>
  <dd>
  Versions: <em>9.5.0.944444</em>
  </dd>

  <dt>mist</dt>
  <dd>
  Versions: 
  </dd>

  <dt>modeller</dt>
  <dd>
  Versions: SVN-i386, SVN-x86_64, 9.10-i386, 9.10-x86_64, 9.10, 9.11-i386, 9.11-x86_64, 9.11, 9.12-i386, 9.12-x86_64, 9.12, 9.13-i386, 9.13-x86_64, 9.13, 9.14-i386, 9.14-x86_64, 9.14, 9.15-i386, 9.15-x86_64, 9.15, 9.16-i386, 9.16-x86_64, 9.16, 9.17-i386, 9.17-x86_64, 9.17, 9.18-i386, 9.18-x86_64, 9.18, 9.19-i386, 9.19-x86_64, 9.19, 9.20-i386, 9.20-x86_64, 9.20, 9.21-i386, 9.21-x86_64, 9.21, <em>SVN</em>
  </dd>

  <dt>modpipe</dt>
  <dd>
  Versions: 2.0.1, 2.0.2, 2.1.0, 2.1.1, 2.1.2, 2.1.3, <em>2.2.0</em>
  </dd>

  <dt>muscle</dt>
  <dd>
  Versions: <em>3.8.31</em>
  </dd>

  <dt>ninja</dt>
  <dd>
  Versions: 1.6.0, <em>1.8.2</em>
  </dd>

  <dt>opencv</dt>
  <dd>
  Versions: <em>3.4.3</em>
  </dd>

  <dt>openeye</dt>
  <dd>
  Versions: <em>2012</em>
  </dd>

  <dt>patch_dock</dt>
  <dd>
  Versions: <em>1.3</em>
  </dd>

  <dt>pcss</dt>
  <dd>
  Versions: 
  </dd>

  <dt>phenix</dt>
  <dd>
  Versions: <em>1.10.1.2155</em>
  </dd>

  <dt>profit</dt>
  <dd>
  Versions: <em>3.1</em>
  </dd>

  <dt>psipred</dt>
  <dd>
  Versions: <em>4.0</em>
  </dd>

  <dt>python/biopython</dt>
  <dd>
  Versions: 1.68, 1.69, <em>1.70</em>
  </dd>

  <dt>python/bottleneck</dt>
  <dd>
  Versions: 1.2.0, <em>1.2.1</em>
  </dd>

  <dt>python/cycler</dt>
  <dd>
  Versions: <em>0.10.0</em>
  </dd>

  <dt>python/dateutil</dt>
  <dd>
  Versions: 1.5, <em>2.6.1</em>
  </dd>

  <dt>python/decorator</dt>
  <dd>
  Versions: 4.0.11, <em>4.2.1</em>
  </dd>

  <dt>python/functools32</dt>
  <dd>
  Versions: <em>3.2.3</em>
  </dd>

  <dt>python/h5py</dt>
  <dd>
  Versions: 2.6.0, 2.7.0, <em>2.7.1</em>
  </dd>

  <dt>python/ihm</dt>
  <dd>
  Versions: <em>0.2</em>
  </dd>

  <dt>python/lru_cache</dt>
  <dd>
  Versions: <em>1.5</em>
  </dd>

  <dt>python/matplotlib</dt>
  <dd>
  Versions: 2.0.0, 2.0.2, <em>2.1.2</em>
  </dd>

  <dt>python/mdp</dt>
  <dd>
  Versions: <em>2.6</em>
  </dd>

  <dt>python/msgpack</dt>
  <dd>
  Versions: <em>0.5.6</em>
  </dd>

  <dt>python/networkx</dt>
  <dd>
  Versions: <em>1.11</em>
  </dd>

  <dt>python/nose</dt>
  <dd>
  Versions: <em>1.3.7</em>
  </dd>

  <dt>python/numexpr</dt>
  <dd>
  Versions: 2.6.2, <em>2.6.4</em>
  </dd>

  <dt>python/numpy</dt>
  <dd>
  Versions: 1.12.0, 1.13.0, <em>1.14.1</em>
  </dd>

  <dt>python/pandas</dt>
  <dd>
  Versions: 0.19.2, 0.20.2, <em>0.22.0</em>
  </dd>

  <dt>python/pkgconfig</dt>
  <dd>
  Versions: <em>1.2.2</em>
  </dd>

  <dt>python/protobuf</dt>
  <dd>
  Versions: <em>2.3.0</em>
  </dd>

  <dt>python/pymc</dt>
  <dd>
  Versions: <em>2.3.6</em>
  </dd>

  <dt>python/pyparsing</dt>
  <dd>
  Versions: <em>2.1.10</em>
  </dd>

  <dt>python/pyrmsd</dt>
  <dd>
  Versions: <em>4.1.git48ab119</em>
  </dd>

  <dt>python/pytz</dt>
  <dd>
  Versions: <em>2016.10</em>
  </dd>

  <dt>python/scikit</dt>
  <dd>
  Versions: 0.12, 0.18.1, <em>0.19.1</em>
  </dd>

  <dt>python/scipy</dt>
  <dd>
  Versions: 0.18.1, 0.19.0, <em>1.0.0</em>
  </dd>

  <dt>python/six</dt>
  <dd>
  Versions: <em>1.10.0</em>
  </dd>

  <dt>python/tables</dt>
  <dd>
  Versions: 3.3.0, <em>3.4.2</em>
  </dd>

  <dt>relion</dt>
  <dd>
  Versions: 1.4, 2.0.6, <em>3.0.git9a02562</em>
  </dd>

  <dt>rosetta</dt>
  <dd>
  Versions: 3.5, <em>3.10</em>
  </dd>

  <dt>sali-libraries</dt>
  <dd>
  Versions: 
  </dd>

  <dt>sali-libraries-i386</dt>
  <dd>
  Versions: 
  </dd>

  <dt>sali-libraries-x86_64</dt>
  <dd>
  Versions: 
  </dd>

  <dt>svm_light</dt>
  <dd>
  Versions: <em>6.0.2</em>
  </dd>

  <dt>swig</dt>
  <dd>
  Versions: <em>3.0.12</em>
  </dd>

  <dt>usearch</dt>
  <dd>
  Versions: 4.0.43, <em>10.0.240</em>
  </dd>

  <dt>web_service</dt>
  <dd>
  Versions: 
  </dd>

  <dt>zdock</dt>
  <dd>
  Versions: <em>3.0.2</em>
  </dd>

</dl>
</div> 


</div> 

<br>
<br>
<br>
_The above information was automatically generated on 2019-06-13 20:17:26._


<style>
dl {
  display: block;
}
dt {
  font-family: Menlo, Monaco, Consolas, "Courier New", monospace;
  font-weight: bold;
  font-size: 120%;
  margin-top: 1ex;
  text-decoration: underline;
}
dt:after {
  content: "";
}
</style>

